device_backends/LogicalNameMapping/src/LogicalNameMappingBackend.cc

```diff
@@ -206,13 +206,16 @@ namespace ChimeraTK {
 
       std::string devName = lnmInfo.deviceName;
 
-      RegisterInfo target_info(lnmInfo.clone()); //Start with a clone of this info as there is not default constructor
-      // In case the devide is not "this" replace it with the real target register info
+      RegisterInfo target_info(lnmInfo.clone()); // Start with a clone of this info as there is not default constructor
+      // In case the device is not "this" replace it with the real target register info
       if(devName != "this") {
         auto cat = _devices.at(devName)->getRegisterCatalogue();
         if(!cat.hasRegister(lnmInfo.registerName)) continue;
         target_info = cat.getRegister(lnmInfo.registerName);
       }
+      else {
```

Comment:
Isn't this redundant? The target_info can been cloned in line 209 from the exact info you are assigning here.

Reply:
In 209, the logical name mapped info is cloned, not the target info.

Reply:
yes, traget_info is never cloned. My wording probably is easy to misunderstand (target info has been cloned .... from).

`lnmInfo` is an item of `_catalogue_mutable` ( we are iterating on it), which is cloned **into** `target_info` in line 209. So the target info is a clone of `lnmInfo`

Line 217 is calling `_catalogue_mutable.getRegisterInfo(lnmInfo.registerName)`, which creates another clone of the info for that register and assigns it to `target_info`. The only difference is that the cloning is done inside the catalogue from the "original" entry, and not from the already existing clone `lnmInfo`. But the content should be identical, thus I think it's redundant.

Reply:
I understood your wording. Still, line 217 obtains the register info of the target register of the register being described by `lnmInfo`. The member names in the register info struct are obviously confusing. `lnmInfo.registerName` is not the name of the register but the name of the target register.